config/http_config.go

```diff
@@ -130,7 +130,7 @@ func (u URL) MarshalJSON() ([]byte, error) {
 	if u.URL != nil {
 		return json.Marshal(u.URL.String())
 	}
-	return nil, nil
+	return []byte("null"), nil
```

Comment:
[uber-nit] This might seem like a crazy idea, but we could save hard-coding the null by marshaling nil:

```suggestion
	return json.Marshal(nil)
```